distributed/scheduler.py

```diff
@@ -6922,7 +6922,10 @@ def workers_to_close(
         if isinstance(key, bytes):
             key = pickle.loads(key)
 
-        groups = groupby(key, self.workers.values())
+        # Long running tasks are typically using a worker_client to
+        # scheduler other tasks. We should never downscale them
+        valid_workers = [ws for ws in self.workers.values() if not ws.long_running]
+        groups = groupby(key, valid_workers)
```

Comment:
this now prohibits downscaling of workers with long-running tasks, e.g. tasks-within-tasks